From: David Cohen <david.a.cohen@intel.com>
Subject: extcon: dc_pwrsrc: always wake_lock() once VBUS is on

Currently dc_pwrsrc driver wake_lock()s only if DCP is detected. It happens
because originally this driver expected USB OTG driver to wake_lock() whenever
an USB connection was detected.
But new scenario is USB OTG driver no longer wake_lock()s. Thus, we move
wake_lock() to happen whenever VBUS is on within dc_pwrsrc driver.

---
 drivers/extcon/extcon-dc-pwrsrc.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- linux.orig/drivers/extcon/extcon-dc-pwrsrc.c
+++ linux/drivers/extcon/extcon-dc-pwrsrc.c
@@ -257,8 +257,6 @@ static int handle_chrg_det_event(struct
 		cable_props.chrg_evt = POWER_SUPPLY_CHARGER_EVENT_CONNECT;
 		cable_props.chrg_type = POWER_SUPPLY_CHARGER_TYPE_USB_DCP;
 		cable_props.ma = DC_XPWR_CHARGE_CUR_DCP;
-		if (!wake_lock_active(&info->wakelock))
-			wake_lock(&info->wakelock);
 	} else {
 		dev_warn(&info->pdev->dev,
 			"disconnect or unknown or ID event\n");
@@ -281,6 +279,8 @@ notify_otg_em:
 		if (wake_lock_active(&info->wakelock))
 			wake_unlock(&info->wakelock);
 	} else {
+		if (!wake_lock_active(&info->wakelock))
+			wake_lock(&info->wakelock);
 		if (notify_otg) {
 			/*
 			 * TODO:close mux path to switch
